props/saber_sa22c_buttons.h

```diff
@@ -0,0 +1,357 @@
+//
+//2 Button: 
+//POWER
+//Activate Muted - fast double click while OFF
+//Activate - short click while OFF
+//Play/Stop Music - hold and release while OFF
+//Turn off blade - hold and wait till blade is off while ON
+//Force Effects - double click while ON
+//Volume UP - short click while OFF and in VOLUME MENU
+//Prev Preset - hold and wait while OFF
+//Color Change mode - hold + toggle AUX while ON
+//AUX
+//Blaster blocks - short click while ON
+//Next Preset - short click while OFF
+//Lockup - hold while ON
+//Drag - hold while ON pointing the blade tip down
+//Enter VOLUME MENU - long click while OFF
+//Volume down - short click while OFF and in VOLUME MENU
+//Battery level - hold while off
+//
+//3 Button: Same as two button except for VOLUME MENU
+//AUX
+//Volume UP - short click while OFF and in VOLUME MENU
+//AUX2
+//Lockup - hold while ON
+//Previous Preset - short click while OFF
+//Volume DOWN - short click while OFF and in VOLUME MENU
+
+#ifndef PROPS_SABER_SA22C_BUTTONS_H
+#define PROPS_SABER_SA22C_BUTTONS_H
+
+#include "prop_base.h"
+
+#undef PROP_TYPE
+#define PROP_TYPE SaberSA22CButtons
+
+// The Saber class implements the basic states and actions
+// for the saber.
+class SaberSA22CButtons : public PropBase {
+public:
+  SaberSA22CButtons() : PropBase() {}
+  const char* name() override { return "SaberSA22CButtons"; }
+
+  bool Event2(enum BUTTON button, EVENT event, uint32_t modifiers) override {
+    switch (EVENTID(button, event, modifiers)) {
+      case EVENTID(BUTTON_POWER, EVENT_PRESSED, MODE_ON):
+      case EVENTID(BUTTON_AUX, EVENT_PRESSED, MODE_ON):
+      case EVENTID(BUTTON_AUX2, EVENT_PRESSED, MODE_ON):
+        if (accel_.x < -0.15) {
+          pointing_down_ = true;
+        } else {
+          pointing_down_ = false;
+        }
+      return true;
+
+#if NUM_BUTTONS == 0
+      case EVENTID(BUTTON_NONE, EVENT_TWIST, MODE_OFF):
+#endif
+      case EVENTID(BUTTON_POWER, EVENT_LATCH_ON, MODE_OFF):
+      case EVENTID(BUTTON_AUX, EVENT_LATCH_ON, MODE_OFF):
+      case EVENTID(BUTTON_AUX2, EVENT_LATCH_ON, MODE_OFF):
+      case EVENTID(BUTTON_POWER, EVENT_CLICK_SHORT, MODE_OFF):
+ #if NUM_BUTTONS == 2
+        if (MODE_VOLUME){
+          STDOUT.println("Volume up");
+          if (dynamic_mixer.get_volume() < VOLUME){
+            current_volume_ += VOLUME * 0.10;
+            dynamic_mixer.set_volume(current_volume_);
+            beeper.Beep(0.5, 2000);
+            STDOUT.print("Current Volume: ");
+            STDOUT.println(dynamic_mixer.get_volume());
+          }
+          else{
+           beeper.Beep(0.5, 3000);
+          }
+        }
+#endif
+        if (!MODE_VOLUME){
+          aux_on_ = false;
+          On();
+        }
+	return true;
+
+      case EVENTID(BUTTON_AUX, EVENT_CLICK_SHORT, MODE_OFF):
+#if NUM_BUTTONS == 2
+        if (MODE_VOLUME){
+          STDOUT.println("Volume Down");
+          if (dynamic_mixer.get_volume() > (0.10 * VOLUME)){
```

Comment:
The volume up/down code appears in several places. Please create helper functions, put the common code in there and call those helper functions from here. (Duplicating code with cut-n-paste is generally not ok.)